lib/update/cluster/builder.go

```diff
@@ -638,43 +637,38 @@ func supportsTaints(gravityPackage loc.Locator) (supports bool, err error) {
 	return defaults.BaseTaintsVersion.Compare(*ver) <= 0, nil
 }
 
-func shouldUpdateEtcd(p planConfig) (updateEtcd bool, installedEtcdVersion string, updateEtcdVersion string, err error) {
+func shouldUpdateEtcd(installedRuntimePackage, updateRuntimePackage loc.Locator, packages pack.PackageService) (etcd *etcdVersion, err error) {
```

Comment:
To determine whether to update etcd, we need to look at the planet packages of the top-level application and not the runtime application as before. This is because the top-level application can override the planet package, in which case we might be looking for the wrong package since the default planet package is not packaged in the resulting tarball if the system container has been replaced in global `systemOptions`.